test/pipe.cc

```diff
@@ -0,0 +1,96 @@
+#include "eventuals/pipe.h"
+
+#include <thread>
+
+#include "eventuals/loop.h"
+#include "eventuals/terminal.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+
+using eventuals::Loop;
+using eventuals::Pipe;
+
+TEST(Pipe, UniqueValue) {
+  Pipe<int> pipe;
+
+  *pipe.Write(1);
+  auto e = [&pipe]() {
+    return pipe.Read()
+        | Loop<int>()
+              .context(/*sum = 0*/ 0)
+              .body([](auto&& sum, auto& repeated, auto&& value) {
+                sum += value;
+                repeated.Done();
+              })
+              .ended([](auto& sum, auto& k) {
+                k.Start(std::move(sum));
+              });
```

Comment:
Let's simplify and use something like `Collect`!

Reply:
I like your idea of using `Collect`. The only thing that is bugging me right now is that the users should always remember to use `*pipe.Close()` in order to make the stream done. If not, we will never get to `.ended` callback in the `Collect()` function.
Actually I've also forgot to call `Close()` :)

Reply:
That's ok that they have to remember to call `Close()`! That's part of the API!